tests/security/test_tls.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from lib.commands import SSHCommandFailed
+
+# This test is designed to verify that TLS connections is secured
+#
+# Requirements:
+# - An XCP-ng host
+
+@pytest.mark.parametrize("protocol", ["tls1", "tls1_1"])
+def test_tls_disabled(host, protocol):
+    """
+    Verifies that specified TLS protocols are disabled on the XCP-ng host.
+    Uses the openssl command via SSH. (Should raise an error)
+    """
+    try:
+        host.ssh(f"openssl s_client -connect localhost:443 -{protocol}")
```

Comment:
does connecting to localhost test the same thing as connecting to the management IP address?

Reply:
Why run openssl from the host? We could run it from the test runner and see what the host answers to outside requests. I think it's acceptable to add openssl to the list of requirements for the test runner. 

We can also likely testing using Python's openssl bindings or client implementation, directly from the test code.